### ISSUE TITLE
`unar` failed to be built

### ISSUE THREAD

**Issue by astrojhgu** [NixOS/nixpkgs#197167]
### Steps To Reproduce
1. `sudo nix-channel --update`
2. `nix-shell -p unar`

### Build log
```
/nix/store/dh600f7nvv102bkqpyiilr6r79qs6qsq-binutils-2.39/bin/ld: Build/DetectorTest.o:(.data.rel.ro+0x10): undefined reference to `__objc_class_n
ame_NSData'
/nix/store/dh600f7nvv102bkqpyiilr6r79qs6qsq-binutils-2.39/bin/ld: Build/DetectorTest.o:(.data.rel.ro+0x18): undefined reference to `__objc_class_n
ame_NSString'
/nix/store/dh600f7nvv102bkqpyiilr6r79qs6qsq-binutils-2.39/bin/ld: libUniversalDetector.a(UniversalDetector.o):(.data.rel.ro+0x0): undefined refere
nce to `__objc_class_name_NSArray'
/nix/store/dh600f7nvv102bkqpyiilr6r79qs6qsq-binutils-2.39/bin/ld: libUniversalDetector.a(UniversalDetector.o):(.data.rel.ro+0x8): undefined refere
nce to `__objc_class_name_NSString'
/nix/store/dh600f7nvv102bkqpyiilr6r79qs6qsq-binutils-2.39/bin/ld: libUniversalDetector.a(UniversalDetector.o):(.data.rel.ro+0x10): undefined refer
ence to `__objc_class_name_NSObject' 
clang-11: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [Makefile.linux:64: DetectorTest] Error 1
make[1]: Leaving directory '/build/source/UniversalDetector'
make: *** [Makefile.linux:75: ../UniversalDetector/libUniversalDetector.a] Error 2
make: *** Waiting for unfinished jobs....
```
The full log is attached as
[unar_build.log](https://github.com/NixOS/nixpkgs/files/9843597/unar_build.log)
### Additional context
This error was found during a system upgrading
```
sudo nixos-rebuild --upgrade
```
and can be reproduced by `nix-shell -p unar`

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.74, NixOS, 22.11 (Raccoon), 22.11pre418157.104e8082de1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.0`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by vs49688** [comment]
Dup of https://github.com/NixOS/nixpkgs/issues/196513. Fixed in https://github.com/NixOS/nixpkgs/pull/196916.

Status: https://nixpk.gs/pr-tracker.html?pr=196916

**Comment by astrojhgu** [comment]
> Dup of #196513. Fixed in #196916.
> 
> Status: https://nixpk.gs/pr-tracker.html?pr=196916

Thanks, and it after `nixos-rebuild boot --upgrade`, this problem has been solved.
